pkg/storer/internal/reserve/reserve_test.go

```diff
@@ -271,6 +285,53 @@ func TestSameChunkAddress(t *testing.T) {
 			t.Fatalf("expected chunk data to not be updated")
 		}
 	})
+
+	t.Run("chunk with different batchID remains untouched", func(t *testing.T) {
+		signer := getSigner(t)
+		batch := postagetesting.MustNewBatch()
+		s1 := soctesting.GenerateMockSocWithSigner(t, []byte("data"), signer)
+		ch1 := s1.Chunk().WithStamp(postagetesting.MustNewFields(batch.ID, 0, 3))
+		batch = postagetesting.MustNewBatch()
+		s2 := soctesting.GenerateMockSocWithSigner(t, []byte("update"), signer)
+		ch2 := s2.Chunk().WithStamp(postagetesting.MustNewFields(batch.ID, 0, 4))
+
+		err = r.Put(ctx, ch1)
```

Comment:
i would add a check here that the two chunks have the same address
also we should have the same unit test for content addressed chunks as well

Reply:
updated